drivers/audio/csaudiork3x/Source/Inc/NewDelete.h

```diff
@@ -0,0 +1,104 @@
+/*++
+
+Copyright (c) Microsoft Corporation All Rights Reserved
+
+Module Name:
+
+NewDelete.h
+
+Abstract:
+
+Declaration of placement new and delete operators.
+--*/
+#pragma once
+
+#ifdef _NEW_DELETE_OPERATORS_
+
+/*****************************************************************************
+* Functions
+*/
+
+/*****************************************************************************
+* ::new()
+*****************************************************************************
+* New function for creating objects with a specified allocation tag and
+* pool type
+*/
+PVOID operator new
+(
```

Comment:
Nit: Technically, C++ requires an `operator delete(ptr, poolType, tag)` to be paired with this, and an `operator delete(ptr, poolType)` for the `operator new(size, poolType)`. 

However, in practice, it would never be used unless C++ exceptions are turned on (and they're not supported in kernel mode), so this is probably fine.

Practically, C++ new/delete overloading requirements are very silly and make me grumpy. The required `operator delete` overload would only be used if the constructor throws an exception and is essentially required to behave exactly like `operator delete(ptr)`, except that it might do additional logging or something.

Reply:
wonder if Microsoft did this just to satisfy C++ requirements, as you mentioned that C++ exceptions are not supported in kernel mode (see https://github.com/microsoft/Windows-driver-samples/blob/main/audio/simpleaudiosample/Source/Inc/NewDelete.h)

Reply:
Microsoft kernel devs are usually not C++ experts. This file is really weird and makes me think they just added stuff until the driver linked, without actually knowing the rules. And then they later removed one of the operator new overloads they were using, but didn't remove the corresponding operator delete overload.